src/commands/new.rs

```diff
@@ -1,75 +1,90 @@
+            input.trim().to_string()
+        };
+    
         let (repo, id) = get_repo_and_id(&index_entry)?;
-
-        self.run_template(repo, id).await
+    
+        self.run_template(repo, id, app_name).await
     }
-
-    async fn run_template(&self, repo: String, id: String) -> anyhow::Result<()> {
+    
+    async fn run_template(&self, repo: String, id: String, app_name: String) -> Result<()> {
         use spin_templates::*;
-
+        
         let tempdir = tempfile::tempdir().unwrap();
-        let template_manager = spin_templates::TemplateManager::in_dir(tempdir.path());
-
+        let manager = spin_templates::TemplateManager::in_dir(tempdir.path());
+    
         let source = TemplateSource::try_from_git(&repo, &None, &crate::spin::version())?;
         let options = InstallOptions::default();
-        template_manager.install(&source, &options, &DiscardingProgressReporter).await?;
-
-        let template = template_manager.get(&id).unwrap().unwrap();
+        manager.install(&source, &options, &DiscardingProgressReporter).await?;
+    
+        let template = match manager.get(&id)? {
+            Some(template) => template,
+            None => return Err(anyhow::anyhow!("Template not found in the repository.")),
+        };
+            
         let options = RunOptions {
             variant: TemplateVariantInfo::NewApplication,
-            name: self.name.clone(),
-            output_path: PathBuf::from(&self.name), // TODO: path safe,
+            name: app_name.clone(),
+            output_path: PathBuf::from(&app_name),
             values: Default::default(),
             accept_defaults: false,
         };
+
         template.run(options).interactive().await
     }
 
-    async fn resolve_selection(&self) -> Result<Option<hub_api::IndexEntry>, dialoguer::Error> {
+    async fn resolve_selection(&self) -> Result<Option<hub_api::IndexEntry>> {
         let entries = hub_api::index().await.unwrap();
         let matches = entries.iter().filter(|e| self.is_match(e)).sorted_by_key(|e| e.title()).collect_vec();
 
-        match matches.len() {
-            0 => {
-                println!("No templates matches your search terms");
-                return Ok(None);
-            }
-            1 => {
-                let index_entry = matches[0].clone();
-                return Ok(Some(index_entry))
-            },
-            _ => {
-                dialoguer::Select::new()
-                    .with_prompt("Several templates match your search. Use arrow keys and Enter to select, or Esc to cancel:")
-                    .items(&matches.iter().map(|e| e.title()).collect_vec())
-                    .interact_opt()?
-                    .map(|idx| Ok(matches[idx].clone()))
-                    .transpose()
-            }
+        if matches.is_empty() {
+            println!("No templates match your search terms");
+            return Ok(None);
+        }
+
+        let selection = dialoguer::Select::new()
```

Comment:
I am not sure why you've removed the "length = 1" case.  This will result in displaying a "select a template" prompt when there's only one thing to select from, which feels awkward.  Is this intentional?